app/services/locale_service/locales.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class LocaleService::Locales # rubocop:disable Style/ClassAndModuleChildren
+  Locale = LocaleService::Locale
+
+  LOCALES = [
+    # english
+    Locale.new(locale: 'english', lang_code: :en, canonical: 'english'),
+
+    # english
+    Locale.new(locale: 'danish', lang_code: :da, canonical: 'dansk'),
+    Locale.new(locale: 'dansk',  lang_code: :da, canonical: 'dansk'),
+
+    # spanish
+    Locale.new(locale: 'espanol', lang_code: :es, canonical: 'espanol'),
+    Locale.new(locale: 'spanish', lang_code: :es, canonical: 'espanol'),
+    Locale.new(locale: 'español', lang_code: :es, canonical: 'espanol'),
+
+    # german
+    Locale.new(locale: 'deutsch', lang_code: :de, canonical: 'deutsch'),
+
+    # finnish
+    Locale.new(locale: 'suomen',  lang_code: :fi, canonical: 'suomen'),
+    Locale.new(locale: 'finnish', lang_code: :fi, canonical: 'suomen'),
+
+    # french
+    Locale.new(locale: 'français', lang_code: :fr, canonical: 'francais'),
+    Locale.new(locale: 'french',   lang_code: :fr, canonical: 'francais'),
+
+    # greek
+    Locale.new(locale: 'ελληνικά', lang_code: :gr, canonical: 'ellenika'),
+    Locale.new(locale: 'greek',    lang_code: :gr, canonical: 'ellenika'),
+
+    # hebrew
+    Locale.new(locale: 'hebrew',   lang_code: :he, canonical: 'ibriyt'),
+    Locale.new(locale: 'עִבְרִית', lang_code: :he, canonical: 'ibriyt'),
+
+    # italian
+    Locale.new(locale: 'italian',  lang_code: :it, canonical: 'italiano'),
+    Locale.new(locale: 'italiano', lang_code: :it, canonical: 'italiano'),
+
+    # portuguese
+    Locale.new(locale: 'portuguese', lang_code: :pt, canonical: 'portugues'),
+    Locale.new(locale: 'portugués',  lang_code: :pt, canonical: 'portugues'),
+
+    # brazilian portuguese
+    Locale.new(locale: 'brazilian portuguese', lang_code: :'pt-br', canonical: 'portugues-brasileiro'),
+    Locale.new(locale: 'português brasileiro', lang_code: :'pt-br', canonical: 'portugues-brasileiro'),
+
+    # swedish
+    Locale.new(locale: 'swedish', lang_code: :sv, canonical: 'svenska'),
+    Locale.new(locale: 'svenska', lang_code: :sv, canonical: 'svenska'),
+
+    # turkish
+    Locale.new(locale: 'turkish', lang_code: :tr, canonical: 'turkce'),
+    Locale.new(locale: 'türkçe',  lang_code: :tr, canonical: 'turkce')
+  ].freeze
+
+  class << self
+    def canonical locale:, lang_code: nil
+      return LOCALES.find { |loc| loc.locale == locale } if lang_code.blank?
```

Comment:
```suggestion
      return LOCALES.find { |locale| locale.locale == locale } if lang_code.blank?
```

Abbrevs are ambiguous, in general.
`loc` usually means `lines of code`, in particular.

Reply:
yeah :grimacing: 

i as shadowing `locale` already from the params, i will think of something clearer